Cookbook/Technical-Documents/FeatureSwitches.md

```diff
@@ -0,0 +1,167 @@
+
+	These feature switches are defined in the `LocalFeatureSwitchesProtocol` and are stored in `UserDefaults`. **These feature switches are supposed to be used only during the development of any new feature so that the code for the feature can be integrated into the `develop` branch continuously.** As they are stored in `UserDefaults` they can be overridden during testing with Settings app. To add a new local feature switch do the following:
+
+	1. Add a new case in the `SettingsKeys` enum for this feature flag in `LocalFeatureSwitches.swift`:
+
+	```swift
+	case isNewFeatureEnabled
+	```	
+	
+	2. Add a new property to the `LocalFeatureSwitchesProtocol` in `LocalFeatureSwitches.swift`:
+
+	```swift
+	static var isNewFeatureEnabled: Bool { get }
+	```
+	
+	3. Add a default implementation in the extension of `LocalFeatureSwitchesProtocol` in `LocalFeatureSwitches.swift`:
+
+	```swift
+    public static var isNewFeatureEnabled: Bool {
+        #if DEBUG || ENABLE_FEATURE_FLAGS
+        return valueFromDefaults(for: .isNewFeatureEnabled)
+        #else
+        return false
+        #endif
+    }
+	```
+	
+	Note: `ENABLE_FEATURE_FLAGS` compilation condition exists so that we are able to disable all the **local feature flags** in the builds that we provide for user testing. This flag should be used **only with local feature switches**.
+	
+	
+	4. Add a new entry in the `Root.plist` for a toggle for this feature. The key name should be the same as the raw value of the case in `SettingsKeys` added before. The entry should be added after the `PSGroupSpecifier` item named `✨ Local Feature Switches ✨` and before `✨ Remote Feature Switches ✨` (this will visually group it with other local feature switches in the Settings app)
+
+		1. if the feature is related to the Babylon app only it is enough to add the entry to the plist located at `Babylon/Brand/babylon/Settings.bundle`
+		2. if the feature is related to all our apps then the same entry should be added to the plist located at `Babylon/Supporting Files/Settings.bundle`
+	
+	```xml
+	<dict>
+		<key>Type</key>
+		<string>PSToggleSwitchSpecifier</string>
+		<key>Title</key>
+		<string>New Feature</string>
+		<key>Key</key>
+		<string>isNewFeatureEnabled</string>
+		<key>DefaultValue</key>
+		<false/>
+	</dict>
+	```
+	
+	**Note that default values are always `false`!**
+
+2. **Application configuration**
+
+	Application configuration, as name implies, exists to specify application specific configuraitons, i.e. if a feature should be enabled or completely disabled for a specific app or if it should use a different content. To define a new application configuration you should add a new property to the `AppConfigurationProtocol` in `AppConfiguration.swift`. If the flag is related to a specific feature then it might be better to define it in the dedicated configuration struct/protocol, i.e. if the flag is related to `Appointments` we have `AppointmentsContentProtocol` for this purpose.
```

Comment:
typo in "configurations"
```suggestion
	Application configuration, as name implies, exists to specify application specific configurations, i.e. if a feature should be enabled or completely disabled for a specific app or if it should use a different content. To define a new application configuration you should add a new property to the `AppConfigurationProtocol` in `AppConfiguration.swift`. If the flag is related to a specific feature then it might be better to define it in the dedicated configuration struct/protocol, i.e. if the flag is related to `Appointments` we have `AppointmentsContentProtocol` for this purpose.
```